diff --git a/modules/access/http/resource.c b/modules/access/http/resource.c
--- a/modules/access/http/resource.c	Mon Feb 12 06:13:08 2018
+++ b/modules/access/http/resource.c	Mon Aug 16 20:22:21 2021
@@ -317,2 +317,17 @@
 
+    if (res -> path)
+    {
+        char * suffix = "\0";
+        for (int i = (int)(strlen(res -> path) - 1); i >= 0; --i)
+        {
+            if (res -> path[i] == '.')
+            {
+                suffix = res -> path + i + 1;
+                break;
+            }
+        }
+        if (strcmp(suffix, "flac") == 0)
+            return strdup("audio/flac");
+    }
+
     const char *type = vlc_http_msg_get_header(res->response, "Content-Type");
